### ISSUE TITLE
Dotnet compile-csc is failing on Ubuntu

### ISSUE THREAD

**Issue by gkhanna79** [dotnet/coreclr#3591]
http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/debug_ubuntu/1011/consoleFull#76848974101778894-4bed-4f57-a720-6efd01df0c42

CC @jaredpar @piotrpMSFT

**Comment by jaredpar** [comment]
The error in listed as the following:

```
Failed to restore BuildTools.
```

How is this a Roslyn issue?

**Comment by gkhanna79** [comment]
@jaredpar I am not saying that is a Roslyn issue - I saw dotnet compile-csc failure and requested you to triage it. Actual issue appears to be the following:

09:34:22 log  : Installing Microsoft.DotNet.BuildTools 1.0.25-prerelease-00177.
09:34:23 error: Microsoft.DotNet.BuildTools 1.0.25-prerelease-00177 is not compatible with DNXCore,Version=v5.0.
09:34:23 error: Some packages are not compatible with DNXCore,Version=v5.0.
09:34:23 error: Microsoft.DotNet.BuildTools 1.0.25-prerelease-00177 is not compatible with DNXCore,Version=v5.0 (ubuntu.14.04-x64).
09:34:23 error: Some packages are not compatible with DNXCore,Version=v5.0 (ubuntu.14.04-x64).
09:34:23 info : Committing restore...
09:34:23 log  : Restore failed in 2845ms.

@piotrpMSFT @ellismg Have you seen this before?

**Comment by ellismg** [comment]
@dagood This failed once during restore but has passed since then.  Have you seen intermittent tool runtime restore failures like this before?

**Comment by dagood** [comment]
I haven't seen it, but I don't restore on Ubuntu often. I agree with @gkhanna79 that the best place to look would be that failed compile, since it fails the publish that I believe would otherwise be copying corerun into the tools dir. (And failing to find corerun is an error later in the log.)

Understanding "Exit Code 139" would be a good start. I wasn't able to find any info on that with a quick search though.

**Comment by jasonwilliams200OK** [comment]
> Understanding "Exit Code 139" would be a good start.

I think this is a segmentation fault in dotnetcli or corerun (SIGSEGV, code 11 actually if we get the last-error from shell --> 139 - 128 (8-bit) = 11). Similar thing happened in corerun here: https://github.com/dotnet/corefx/pull/6716#issuecomment-193388192

**Comment by ellismg** [comment]
I agree with @jasonwilliams200OK.  Unfortunately we don't have a good way to get dumps from the CI system.  @gkhanna79 Do you know someone who might be able to follow up on trying to get this to repro locally?

**Comment by gkhanna79** [comment]
@ellismg I am not aware of a process to get crash dumps from CI. Are you suggesting that the PR @jasonwilliams200OK  pointed to is the fix?
